src/components/side_panel/chart/line_bar_pie_panel/config_panel.ts

```diff
@@ -34,6 +38,9 @@ export class LineBarPieConfigPanel extends Component<Props, SpreadsheetChildEnv>
   setup() {
     this.dataSeriesRanges = this.props.definition.dataSets;
     this.labelRange = this.props.definition.labelRange;
+    if (!this.isDatasetInvalid && !this.isLabelInvalid) {
+      this.state.headerIndex = this.calculateHeaderPosition();
+    }
```

Comment:
I still keep the state here because two `dispatchResult` are also states. If I don't use state for header index, when the dataset/label is actually invalid, the states of dispatch results are not updated immediately (which leads to valid judgement), so `calculateHeadPosition()` will throw error and the whole app crashes. I don't know whether there is any way better than states 🤔